tests/integration/test_memory_management.py

```diff
@@ -0,0 +1,62 @@
+import pytest
+
+from autogpt.chat import save_memory_trimmed_from_context_window
+from autogpt.config import Config
+from autogpt.memory import get_memory
+from tests.utils import requires_api_key
+
+
+@pytest.fixture
+def message_history_fixture():
+    test = """{
+    \"thoughts\": {
+        \"text\": \"thoughts\",
+        \"reasoning\": \"reasoning\",
+        \"plan\": \"plan\",
+        \"criticism\": \"criticism\",
+        \"speak\": \"speak\"
+    },
+    \"command\": {
+        \"name\": \"google\",
+        \"args\": {
+            \"query\": \"google_query\"
+        }
+    }
+}"""
+    return [{"content": test}, {"content": "Command Result: Important Information."}]
+
+
+@pytest.fixture
+def expected_permanent_memory():
+    return """Assistant Reply: {
```

Comment:
Don't we want `Assistant Reply` in quotes?

Maybe we should `-> str` in the function def'n line, to remind outselves we're returning a string not a dict. And the same trick of writing out a dict and using `json.dumps` would be clean here.

Reply:
Note, as I don't quite understand what all this is doing, please ignore me if that above idea was junk.